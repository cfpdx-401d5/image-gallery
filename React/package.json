{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
