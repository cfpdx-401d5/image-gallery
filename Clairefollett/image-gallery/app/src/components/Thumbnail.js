import React from 'react';

export default function Thumbnail(props) {
    const thumbnail = props.puppies.map(puppies => {
        return (
            <ThumbnailDisplay key={puppies._id} puppies={puppies} onDelete={props.onDelete}/>
        );
    });
    return (
      <ul>{thumbnail}</ul>
    );
}

function ThumbnailDisplay(props) {
  return (
    <div className="clearfix">
      <li>
        <p height='20px' width='200'>title: {props.puppies.title}</p>
        <img className='thumbnail' src={props.puppies.url} alt='' height='100' width='100' />
        <div className="delete"><button className="shadow animate orange" onClick={() => props.onDelete(props.puppies._id)}>Delete</button></div>
      </li>
    </div>
  );
}
