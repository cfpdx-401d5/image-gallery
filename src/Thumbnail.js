import React, { PropTypes } from 'react';

function Thumbnail(props) {
  let altText = props.title + ' thumbnail';

  return (
    <figure>
      <img className='thumbnail' src={props.url} alt={altText} />
      <figcaption>{props.title}</figcaption>
    </figure>
  );
}
Thumbnail.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default Thumbnail;
