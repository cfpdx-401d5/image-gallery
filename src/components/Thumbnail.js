import React from 'react';

export default function Thumbnail(props) {
    return (
        <div>
            <img alt={props.image.title} src={props.image.url} height='100px' width='100px' />
            <h4>{props.image.title}</h4>
        </div>
    );
}