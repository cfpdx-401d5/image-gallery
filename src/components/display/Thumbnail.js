import React from 'react';

export default function ThumbnailDisplay(props) {
    return (
      <div>
        <h1>Title: {props.resource.title}</h1>
        <img className='thumbnail' src={props.resource.url} alt='Cute Bunny' />
        <button onClick={() => props.onDelete(props.resource.id)}>delete</button>
      </div>
    );
};
