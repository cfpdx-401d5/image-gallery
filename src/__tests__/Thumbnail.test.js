/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Thumbnail from '../Thumbnail';

describe('Thumbnail component', function () {
    it('renders the Thumbnail component', () => {
        const rendered = shallow(<Thumbnail url='_url' title='_title'/>);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});
