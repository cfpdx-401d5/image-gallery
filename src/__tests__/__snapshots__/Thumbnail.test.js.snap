exports[`Thumbnail component renders the Thumbnail component 1`] = `
<figure>
  <img
    alt="_title thumbnail"
    className="thumbnail"
    src="_url" />
  <figcaption>
    _title
  </figcaption>
</figure>
`;
